### PR TITLE
Speed up and calculate dist parameter for tigmint-long

### PR DESCRIPTION
**Pull request by janetxinli**

### CODE REVIEW

File: bin/tigmint-molecule [hunk 1]
```diff
@@ -62,7 +61,7 @@ class MolecIdentifier:
     """Group molecules into barcodes"""
 
     def __init__(self):
-        """Constructor, identifies molecules based on inter-arrival time threshold."""
+        """Constructor"""
```
Comment (lcoombe) [inline bin/tigmint-molecule:64]:
```suggestion
        """Constructor"""
```
I think we could keep this comment in? We're still using inter-arrival distances to infer molecules, even though we aren't saving the inter-arrivals anymore?

File: bin/tigmint-molecule [hunk 2]
```diff
@@ -273,7 +267,29 @@ class MolecIdentifier:
             "-s", "--size", dest="min_size", type=int, default=2000,
             help="Minimum molecule size [2000]",
             metavar="N")
+        parser.add_argument(
+            "-p", "--params", dest="param_file", type=str)
+        
         self.opt = parser.parse_args()
+        
+        # Get read length p5 from file if max_dist=1
```
Comment (lcoombe) [inline bin/tigmint-molecule:275]:
```suggestion
        # Get read length p5 from file if max_dist=-1
```
Or `auto`, depending on what you think about my earlier comment

**Review by lcoombe — COMMENTED** [review]
This is looking great, Janet! Just a few comments.

File: README.md [hunk 3]
```diff
@@ -121,7 +121,7 @@ tigmint-make metrics draft=myassembly reads=myreads ref=GRCh38 G=3088269832
 To run Tigmint with long reads in fasta or fastq format (`reads.fa.gz` or `reads.fq.gz`) on the draft assembly `draft.fa`:
 
 ```sh
-tigmint-make tigmint-long draft=myassembly reads=myreads span=auto G=genomesize
+tigmint-make tigmint-long draft=myassembly reads=myreads span=auto G=genomesize dist=-1
```
[the remark this answers is not in the document]
Comment (janetxinli) [reply README.md:124]: Yes of course!

**Comment by janetxinli** [comment]
Alrighty, just made those changes! 😄

File: README.md [hunk 4]
```diff
@@ -118,10 +118,10 @@ tigmint-make metrics draft=myassembly reads=myreads ref=GRCh38 G=3088269832
 ```
 ***
 
-To run Tigmint with long reads in fasta or fastq format (`reads.fa.gz` or `reads.fq.gz`) on the draft assembly `draft.fa`:
+To run Tigmint with long reads in fasta or fastq format (`myreads.fa.gz` or `myreads.fq.gz`) on the draft assembly `draft.fa`:
```
Comment (lcoombe) [inline README.md:121]:
```suggestion
To run Tigmint with long reads in fasta or fastq format (`myreads.fa.gz` or `myreads.fq.gz`) on the draft assembly `myassembly.fa`:
```

**Review by lcoombe — APPROVED** [review]
Looks good after that last small README fix! Great work Janet :)

**Comment by janetxinli** [comment]
Omg sorry, glad you caught that haha I swear I looked through like three times, can't believe I missed the one right next to the first change 🤦

**Comment by lcoombe** [comment]
Hahaha it's much easier for someone with fresh eyes to see these small things!!

**Pull request merged**